lib/google_crawler/search/keyword_file.ex

```diff
@@ -0,0 +1,38 @@
+defmodule GoogleCrawler.Search.KeywordFile do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  embedded_schema do
+    field :file, :map
+  end
+
+  @accept_file_ext ~w(.csv)
+
+  def changeset(keyword_file, attrs \\ %{}) do
+    keyword_file
+    |> cast(attrs, [:file])
+    |> validate_required([:file])
+    |> validate_file_ext()
+  end
+
+  def parse!(file_path, "text/csv") do
+    file_path
+    |> File.stream!()
+    |> CSV.decode!()
+  end
+
+  def parse!(_file_path, _unexpected_mime_type) do
+    raise GoogleCrawler.Errors.FileNotSupportedError,
+      message: "File with this extension is not supported"
+  end
+
+  defp validate_file_ext(changeset) do
+    validate_change(changeset, :file, fn :file, file ->
+      if Enum.member?(@accept_file_ext, Path.extname(file.filename)) do
+        []
+      else
+        [file: "is not supported"]
+      end
```

Comment:
I rechecked this one. I found something interesting 📝 

When using the `validate_change`, seems like it will only validate when the value is not nil - https://hexdocs.pm/ecto/Ecto.Changeset.html#validate_change/3. So after removing it out, there is one case that if the file is nil, it will still validate and the error occurs when `file.filename` because the file is `nil`. For this, we might need to add more condition/clause function to handle this case 👍